woodwork/table_accessor.py

```diff
@@ -145,6 +145,30 @@ def __getattr__(self, attr):
         else:
             raise AttributeError(f"Woodwork has no attribute '{attr}'")
 
+    def __getitem__(self, key):
+        if isinstance(key, list):
+            index = pd.Index(key)
+            columns = self._dataframe.columns
+            in_columns = index.isin(columns).all()
+
+            if not in_columns:
+                diff = list(index.difference(columns).sort_values())
+                raise KeyError(f"Column(s) '{diff}' not found in DataTable")
+
+            df = self._dataframe[key]
+            schema = self._schema._get_subset_schema(key)
+            df.ww.init(schema=schema)
+            return df
+
+        if key not in self._dataframe:
+            raise KeyError(f"Column with name '{key}' not found in DataTable")
+
+        series = self._dataframe[key]
+        column = self._dataframe.ww.columns[key].copy()
```

Comment:
We can get the column with `self._schema.columns[key]` (though it's super interesting how we can access the accessor's attrs within the accessor like that--kind of mind bending)